keanu-project/src/main/java/io/improbable/keanu/vertices/intgr/nonprobabilistic/operators/unary/IntegerReshapeVertex.java

```diff
@@ -0,0 +1,22 @@
+package io.improbable.keanu.vertices.intgr.nonprobabilistic.operators.unary;
+
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.intgr.IntegerVertex;
+
+public class IntegerReshapeVertex extends IntegerUnaryOpVertex {
+
+    private int[] proposedShape;
+
+    public IntegerReshapeVertex(IntegerVertex inputVertex, int... proposedShape) {
+        super(inputVertex.getShape(), inputVertex);
```

Comment:
Output shape is the proposed shape, not the inputVertexShape.

Reply:
nice, changed